packages/@aws-cdk/aws-rds/test/cluster.test.ts

```diff
@@ -762,9 +822,101 @@ describe('cluster', () => {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const vpc = new ec2.Vpc(stack, 'VPC');
+    const cluster = new DatabaseCluster(stack, 'Database', {
+      engine: DatabaseClusterEngine.AURORA_MYSQL,
+      credentials: { username: 'admin' },
+      instanceProps: {
+        instanceType: ec2.InstanceType.of(ec2.InstanceClass.BURSTABLE2, ec2.InstanceSize.SMALL),
+        vpc,
+      },
+    });
+
+    // WHEN
+    cluster.addRotationSingleUser();
+
+    // THEN
+    expect(() => {
+      cluster.addRotationSingleUser({ secret: cluster.secret });
+    }).toThrow(/A single user rotation for master secret is already added to this cluster/);
+  });
+
+  test('add single user rotation for two additional secrets', () => {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const vpc = new ec2.Vpc(stack, 'VPC');
+    const cluster = new DatabaseCluster(stack, 'Database', {
+      engine: DatabaseClusterEngine.AURORA_MYSQL,
+      credentials: { username: 'admin' },
+      instanceProps: {
+        instanceType: ec2.InstanceType.of(ec2.InstanceClass.BURSTABLE2, ec2.InstanceSize.SMALL),
+        vpc,
+      },
+    });
+    const secret1 = new DatabaseSecret(stack, 'MySecret1', { username: 'user1' });
+    secret1.attach(cluster);
+    const secret2 = new DatabaseSecret(stack, 'MySecret2', { username: 'user2' });
+    secret2.attach(cluster);
+
+    // WHEN
+    cluster.addRotationSingleUser({ secret: secret1 });
+    cluster.addRotationSingleUser({ secret: secret2 });
+
+    // THEN
+    expect(stack).toHaveResource('AWS::SecretsManager::RotationSchedule', {
+      SecretId: {
+        Ref: 'MySecret1E18D875B',
+      },
+      RotationLambdaARN: {
+        'Fn::GetAtt': [
+          'MySecret1RotationSingleUser7D84B21E',
+          'Outputs.RotationLambdaARN',
+        ],
+      },
+    });
+    expect(stack).toHaveResource('AWS::SecretsManager::RotationSchedule', {
+      SecretId: {
+        Ref: 'MySecret225E18D70',
+      },
+      RotationLambdaARN: {
+        'Fn::GetAtt': [
+          'MySecret2RotationSingleUser2903C638',
+          'Outputs.RotationLambdaARN',
+        ],
+      },
+    });
+  });
 
+  test('throws when trying to add single user rotation for additional secret multiple times', () => {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const vpc = new ec2.Vpc(stack, 'VPC');
+    const cluster = new DatabaseCluster(stack, 'Database', {
+      engine: DatabaseClusterEngine.AURORA_MYSQL,
+      credentials: { username: 'admin' },
+      instanceProps: {
+        instanceType: ec2.InstanceType.of(ec2.InstanceClass.BURSTABLE2, ec2.InstanceSize.SMALL),
+        vpc,
+      },
+    });
+    const secret = new DatabaseSecret(stack, 'MySecret', {
+      username: 'myuser',
+    });
+    secret.attach(cluster);
+
+    // WHEN
+    cluster.addRotationSingleUser({ secret });
 
+    // THEN
+    expect(() => {
+      cluster.addRotationSingleUser({ secret });
```

Comment:
I wonder - should this really throw? Or should the call be idempotent, and just the second one do nothing?

Reply:
I personally don't like silently handling programmers' mistakes in the code. But your call!

Reply:
Why do you think it's a mistake though?

Reply:
It'll happen only if they forget that it's already added. As far as I can tell, there's no valid scenario in which one might intentionally wanna add rotation for the secret multiple times. Unless I'm missing something.

Reply:
Still not sure why that is a mistake though.

To me, this method is like a setter. I don't think it's a mistake to write:

```ts
someClass.setValue('myvalue');
someClass.setValue('myvalue');
```

I think it would be pretty surprising if that second call threw an exception saying something like "Value is already 'myvalue'".

Same situation here - if I want to call `cluster.addRotationSingleUser({ secret })`, it shouldn't matter whether that `secret` was already set to rotate, or not - that's an implementation detail. The important thing is that it's for sure set to rotate after my call 🙂.

Reply:
It's a little different in my opinion: almost always a setter is considered an idempotent operation (with a relatively strict definition of idempotency). But not adding a rotation to the stack. That'll make more sense when looking at the verb of the methods; set vs add (intuitively one might think set is idempotent and add is not). Besides if we let users to call `addRotationSingleUser` multiple times without throwing any error, that'll make the library inconsistent; the old implementation throws an error is you call `addRotationSingleUser()` multiple times.

Reply:
Pretty sure `Set.add()` would not agree with you on idempotency 🙂. But the consistency of the module is a fair point. Let's leave the throwing in (maybe we'll revisit this decision in a later PR).